include/rospack/rospack.h

```diff
@@ -194,18 +202,32 @@ class ROSPACK_DECL Rosstackage
     std::string tag_;
     bool quiet_;
     std::vector<std::string> search_paths_;
+#if BOOST_VERSION <= 106500
     std::tr1::unordered_map<std::string, std::vector<std::string> > dups_;
     std::tr1::unordered_map<std::string, Stackage*> stackages_;
+#else
+    boost::unordered_map<std::string, std::vector<std::string> > dups_;
+    boost::unordered_map<std::string, Stackage*> stackages_;
+#endif
     Stackage* findWithRecrawl(const std::string& name);
     void log(const std::string& level, const std::string& msg, bool append_errno);
     void clearStackages();
     void addStackage(const std::string& path);
+#if BOOST_VERSION <= 106500
     void crawlDetail(const std::string& path,
                      bool force,
                      int depth,
                      bool collect_profile_data,
                      std::vector<DirectoryCrawlRecord*>& profile_data,
                      std::tr1::unordered_set<std::string>& profile_hash);
+#else
+    void crawlDetail(const std::string& path,
+                     bool force,
+                     int depth,
+                     bool collect_profile_data,
+                     std::vector<DirectoryCrawlRecord*>& profile_data,
+                     boost::unordered_set<std::string>& profile_hash);
+#endif
```

Comment:
Please narrow the scope of this ifdef block to the line which is different.

Same below.